crates/next-dev-tests/tests/integration.rs

```diff
@@ -207,49 +216,152 @@ async fn create_browser(is_debugging: bool) -> Result<(Browser, JoinHandle<()>)>
 }
 
 async fn run_browser(addr: SocketAddr) -> Result<JestRunResult> {
-    if *DEBUG_BROWSER {
-        run_debug_browser(addr).await?;
-    }
-
-    run_test_browser(addr).await
-}
-
-async fn run_debug_browser(addr: SocketAddr) -> Result<()> {
-    let (browser, handle) = create_browser(true).await?;
-    let page = browser.new_page(format!("http://{}", addr)).await?;
-
-    let run_tests_msg =
-        "Entering debug mode. Run `await __jest__.run()` in the browser console to run tests.";
-    println!("\n\n{}", run_tests_msg);
-    page.evaluate(format!(
-        r#"console.info("%cTurbopack tests:", "font-weight: bold;", "{}");"#,
-        run_tests_msg
-    ))
-    .await?;
-
-    // Wait for the user to close the browser
-    handle.await?;
-
-    Ok(())
+    run_test_browser(addr, *DEBUG_BROWSER).await
 }
 
-async fn run_test_browser(addr: SocketAddr) -> Result<JestRunResult> {
-    let (browser, _) = create_browser(false).await?;
+async fn run_test_browser(addr: SocketAddr, is_debugging: bool) -> Result<JestRunResult> {
+    let (browser, mut handle) = create_browser(is_debugging).await?;
 
     // `browser.new_page()` opens a tab, navigates to the destination, and waits for
     // the page to load. chromiumoxide/Chrome DevTools Protocol has been flakey,
     // returning `ChannelSendError`s (WEB-259). Retry if necessary.
     let page = retry_async(
         (),
-        |_| browser.new_page(format!("http://{}", addr)),
+        |_| browser.new_page("about:blank"),
         5,
         Duration::from_millis(100),
     )
     .await
     .context("Failed to create new browser page")?;
 
-    let value = page.evaluate("globalThis.waitForTests?.() ?? __jest__.run()");
-    Ok(value.await?.into_value()?)
+    page.execute(AddBindingParams::new("READY")).await?;
+
+    let mut errors = page
+        .event_listener::<EventExceptionThrown>()
+        .await
+        .context("Unable to listen to exception events")?;
+    let mut binding_events = page
+        .event_listener::<EventBindingCalled>()
+        .await
+        .context("Unable to listen to binding events")?;
+    let mut console_events = page
+        .event_listener::<EventConsoleApiCalled>()
+        .await
+        .context("Unable to listen to console events")?;
+    let mut network_response_events = page
+        .event_listener::<EventResponseReceived>()
+        .await
+        .context("Unable to listen to response received events")?;
+
+    page.evaluate_expression(format!("window.location='http://{addr}'"))
```

Comment:
is there a reason for this change and for not using `page.goto(format!("http://{}", addr))`?

Reply:
I copied that from the benchmarks. The difference is that this doesn't wait for the load event to happen. I guess it doesn't matter that much here.